find-git.wake

```diff
@@ -0,0 +1,35 @@
+# Find a git repository.
+# A valid repository must have a directory named '.git' within it.
+# If "x/y/z/.git" is found, return "x/y/z"
+global def findGitRepository name =
+  require Pass tree = repositoryTree Unit
+  def hits = getPairFirst (tequal name tree)
+  match hits
+    Nil    = Fail (makeError "git repository '{name}' not found")
+    x, Nil = Pass x
+    _      = Fail (makeError "more than one git repository named '{name}': {catWith ", " hits}.")
+
+
+# Use the name as the fallback path if the repository is not found
+global def findGitRepositoryWithFallback name =
+  match (findGitRepository name)
+    Pass path = path
+    Fail _    = name
```

Comment:
This essentially reproduces the function `getWhenFail`.
ie: findGitRepository repo | getWhenFail repo